src/Type/Php/SimpleXMLElementXpathMethodReturnTypeExtension.php

```diff
@@ -0,0 +1,66 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\NeverType;
+use PHPStan\Type\StaticType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use PHPStan\Type\TypeUtils;
+
+class SimpleXMLElementXpathMethodReturnTypeExtension implements \PHPStan\Type\DynamicMethodReturnTypeExtension
+{
+
+	public function getClass(): string
+	{
+		return \SimpleXMLElement::class;
+	}
+
+	public function isMethodSupported(MethodReflection $methodReflection): bool
+	{
+		return $methodReflection->getName() === 'xpath';
+	}
+
+	public function getTypeFromMethodCall(MethodReflection $methodReflection, MethodCall $methodCall, Scope $scope): Type
+	{
+		if (!isset($methodCall->args[0])) {
+			return ParametersAcceptorSelector::selectSingle($methodReflection->getVariants())->getReturnType();
+		}
+
+		$argType = $scope->getType($methodCall->args[0]->value);
+
+		$xmlElement = new \SimpleXMLElement('<foo />');
+
+		$result = null;
+		foreach (TypeUtils::getConstantStrings($argType) as $constantString) {
+			set_error_handler(null);
+			$newResult = $xmlElement->xpath($constantString->getValue());
```

Comment:
What about `@` here?

Reply:
It seems to work, but some randoms tests are failing...